google/cloud/spanner/sql_partition.h

```diff
@@ -57,74 +100,34 @@ class SqlPartition {
 
   /**
    * Accessor for the `SqlStatement` associated with this `SqlPartition`.
-   * @return SqlStatement
    */
-  SqlStatement const& sql_statement() const;
+  SqlStatement const& sql_statement() const { return sql_statement_; }
 
  private:
   friend class SqlPartitionTester;
-  friend SqlPartition internal::MakeSqlPartition(std::string transaction_id,
-                                                 std::string session_id,
-                                                 std::string partition_token,
-                                                 SqlStatement sql_statement);
-  friend std::string SerializeSqlPartition(SqlPartition const& sql_partition);
-  friend google::cloud::StatusOr<SqlPartition> DeserializeSqlPartition(
+  friend SqlPartition internal::MakeSqlPartition(
+      std::string const& transaction_id, std::string const& session_id,
+      std::string const& partition_token, SqlStatement const& sql_statement);
+  friend StatusOr<std::string> SerializeSqlPartition(
+      SqlPartition const& sql_partition);
+  friend StatusOr<SqlPartition> DeserializeSqlPartition(
       std::string const& serialized_sql_partition);
 
   explicit SqlPartition(std::string transaction_id, std::string session_id,
                         std::string partition_token,
                         SqlStatement sql_statement);
 
   // Accessor methods for use by friends.
-  std::string const& partition_token() const;
-  std::string const& session_id() const;
-  std::string const& transaction_id() const;
+  std::string const& partition_token() const { return partition_token_; }
+  std::string const& session_id() const { return session_id_; }
+  std::string const& transaction_id() const { return transaction_id_; }
 
   std::string transaction_id_;
   std::string session_id_;
   std::string partition_token_;
   SqlStatement sql_statement_;
 };
 
-/**
- * Serializes an instance of `SqlPartition` for transmission to another process.
- * @param sql_partition - instance to be serialized.
- * @return `std::string`
- *
- * @par Example:
- *
- * @code
- * spanner::SqlStatement stmt("select * from Albums");
- * std::vector<spanner::SqlPartition> partitions =
- *   spanner_client.PartitionSql(stmt);
- * for (auto const& partition : partitions) {
- *   SendToRemoteMachine(spanner::SerializeSqlPartition(partition));
- * }
- * @endcode
- */
-std::string SerializeSqlPartition(SqlPartition const& sql_partition);
```

Comment:
Any reason these declarations moved to the top of the file? I'd generally expect them to come after the `SqlPartition` class.

Reply:
For most users of the code, the serialize/deserialize methods would be the main reasons to look at the contents of this file. The SqlPartition class has few useful public members, and the SqlPartition class needed to be forward declared to allow for the friend factory function definition in the class. I felt it would be more convenient to arrange things this way.

Reply:
That makes sense. In general, we prefer a (somewhat) [consistent ordering](https://google.github.io/styleguide/cppguide.html#Declaration_Order) for declarations rather than ordering by usage. But this whole file is written pretty clearly, so there's probably no issue either way. Happy to go with your preference here.